tensorflow/core/platform/env.cc

```diff
@@ -293,6 +304,9 @@ bool Env::LocalTempFilename(string* filename) {
     pthread_threadid_np(nullptr, &tid64);
     int32 tid = static_cast<int32>(tid64);
     int32 pid = static_cast<int32>(getpid());
+#elif defined(__FreeBSD__)
+    int32 tid = static_cast<int32>((long) pthread_self());
```

Comment:
do we need the extra cast to long here?
Cant we just rely on the else branch a few lines below?

Reply:
Hmm, if its not casted to long first, this compile error appears:
>error: static_cast from 'pthread_t' (aka 'pthread *') to 'int32' (aka 'int') is not allowed
>int32 tid = static_cast<int32>(pthread_self());

I'm no c coder so don't really know if there is a more elegant solution to this issue :)

Reply:
I see. Then let's add a comment stating this, then I am OK with this.